app/src/main/java/com/keylesspalace/tusky/util/NotificationHelper.java

```diff
@@ -104,75 +139,168 @@ public static void make(final Context context, Notification body, AccountEntity
-                        .transform(new RoundedTransformation(20))
-                        .get();
-            } catch (IOException e) {
-                Log.d(TAG, "error loading account avatar", e);
-                accountAvatar = BitmapFactory.decodeResource(context.getResources(), R.drawable.avatar_default);
-            }
+            NotificationCompat.Action quickReplyAction =
+                    new NotificationCompat.Action.Builder(R.drawable.ic_reply_24dp,
+                            context.getString(R.string.action_quick_reply), quickReplyPendingIntent)
+                            .addRemoteInput(replyRemoteInput)
+                            .build();
 
-            builder.setLargeIcon(accountAvatar);
+            builder.addAction(quickReplyAction);
 
-        } else {
+            PendingIntent composePendingIntent = getStatusReplyIntent(COMPOSE_ACTION, context, body, account);
+
+            NotificationCompat.Action composeAction =
+                    new NotificationCompat.Action.Builder(R.drawable.ic_reply_24dp,
+                            context.getString(R.string.action_compose_shortcut), composePendingIntent)
+                            .build();
+
+            builder.addAction(composeAction);
+        }
+
+        builder.setSubText(account.getFullName());
+        builder.setVisibility(NotificationCompat.VISIBILITY_PRIVATE);
+        builder.setCategory(NotificationCompat.CATEGORY_SOCIAL);
+        builder.setOnlyAlertOnce(true);
+
+        // Summary
+        // =======
+        final NotificationCompat.Builder summaryBuilder = newNotification(context, body, account, true);
+
+        if (currentNotifications.length() != 1) {
             try {
                 String title = context.getString(R.string.notification_title_summary, currentNotifications.length());
                 String text = joinNames(context, currentNotifications);
-                builder.setContentTitle(title)
+                summaryBuilder.setContentTitle(title)
                         .setContentText(text);
             } catch (JSONException e) {
                 Log.d(TAG, Log.getStackTraceString(e));
             }
         }
 
-        builder.setSubText(account.getFullName());
+        summaryBuilder.setSubText(account.getFullName());
+        summaryBuilder.setVisibility(NotificationCompat.VISIBILITY_PRIVATE);
+        summaryBuilder.setCategory(NotificationCompat.CATEGORY_SOCIAL);
+        summaryBuilder.setOnlyAlertOnce(true);
+        summaryBuilder.setGroupSummary(true);
 
-        builder.setVisibility(NotificationCompat.VISIBILITY_PRIVATE);
-        builder.setCategory(NotificationCompat.CATEGORY_SOCIAL);
+        NotificationManagerCompat notificationManager = NotificationManagerCompat.from(context);
 
-        builder.setOnlyAlertOnce(true);
+        //noinspection ConstantConditions
+        notificationManager.notify(Integer.valueOf(body.getId()), builder.build());
```

Comment:
I don't think it is a good idea to use Mastodon notification ids as Android notification ids.
- they might not be unique since we support multiple accounts (extremly unlikely, but still)
- they might be way bigger than Integer.MAX_VALUE

Reply:
@connyduck I don't like that solution too. Do you have any other proposals?